docs/DeveloperGuide.md

```diff
@@ -1,37 +1,300 @@
+- Add expenditure component.
+- Add credit card component.
+- Add income component.
+
+The sequence diagram below shows the interactions of when an `AddCommand` is parsed.  
+
+![add_command_sequence_diagram](images/Add_Command_Sequence_Diagram.png)  
+Fig 7 - AddCommand Sequence Diagram  
+
+Below is an example scenario showing how the AddCommand behaves at each step.
+1. The `Parser` component parses user input and returns the new `AddCommand` object to the
+   `MindMyMoney`.
+2. `AddCommand` instantiates `addInput`, `expenditureList`, `creditCardList`, `incomeList`.
+3. The application invokes `AddCommand.executeCommand()` to execute user instruction.
+4. If user input contains credit card flag, the application executes `AddCommand.addCreditCard()`.
+5. Else if user input contains income flag, the application executes `AddCommand.addIncome()`.
+6. Else the application executes `AddCommand.addExpenditure()`.
+7. The application then returns to the Parser component.
+
+#### Add Credit Card Command
+MindMyMoney allow users to track their different credit cards. A user can add a new credit card through the 
+`AddCommand.addCreditCard()` command by specifying the credit card name, cash back, card limit and balance.
+
+##### Current Implementation
+The sequence diagram below shows the interactions of different subcomponents of the system when adding a credit card
+to the list.
+
+![add_credit_card_command_sequence_diagram](images/Add_Credit_Card_Command_Sequence_Diagram.png)   
+Fig 8 - Add Credit Card Command Sequence Diagram  
+
+The `AddCommand.addCreditCard()` command is facilitated by the `AddCommand`. By running the command `add` with its relevant flags,
+`Parser` will construct a new `AddCommand` which will be used to execute users input.
+
+1. During the execution, `AddCommand.addCreditCard()` will parse through user input to obtain the `CREDIT_CARD_NAME`, `CATEGORY`,
+   `CASH_BACK`, `CARD_LIMIT` and `CARD_BALANCE` fields.
+2. Once all the fields are obtained, `AddCommand.addCreditCard()` object instantiates a new `CreditCard` 
+   object with the aforementioned 5 fields and adds them into the `CreditCardList`.
+3. The `AddCommand.addCreditCard()` object prints a list to show the user what it has saved.
+4. The `AddCommand.addCreditCard()` returns to `AddCommand`.
+
+#### Add Expenditure Command
+A key functionality of MindMyMoney is the ability to add and track user expenditure. A user can add in a new expenditure
+through the `AddCommand.addExpenditure()` command by specifying the payment method, the category, the description of the item, the cost of the item and the date of 
+purchase.
+
+##### Current Implementation
+The sequence diagram below shows the interactions of different subcomponents of the system when adding an expenditure
+to the list.  
+
+![add_expenditure_command_sequence_diagram](images/Add_Expenditure_Command_Sequence_Diagram.png)  
```

Comment:
for the print added expenditure, should it be calling the print()/equivilent print method?